test/sortable.e2e.extraElements.spec.js

```diff
@@ -0,0 +1,716 @@
+'use strict';
+
+describe('uiSortable', function() {
+
+  beforeEach(module(function($compileProvider) {
+    if (typeof $compileProvider.debugInfoEnabled === 'function') {
+      $compileProvider.debugInfoEnabled(false);
+    }
+  }));
+
+  // Ensure the sortable angular module is loaded
+  beforeEach(module('ui.sortable'));
+  beforeEach(module('ui.sortable.testHelper'));
+
+  var EXTRA_DY_PERCENTAGE, listContent, listInnerContent;
+
+  beforeEach(inject(function (sortableTestHelper) {
+    EXTRA_DY_PERCENTAGE = sortableTestHelper.EXTRA_DY_PERCENTAGE;
+    listContent = sortableTestHelper.listContent;
+    listInnerContent = sortableTestHelper.listInnerContent;
+  }));
+
+  describe('Drag & Drop simulation, when there are extra elements', function() {
+
+    var host;
+
+    beforeEach(inject(function() {
+      host = $('<div id="test-host"></div>');
+      $('body').append(host);
+    }));
+
+    afterEach(function() {
+      host.remove();
+      host = null;
+    });
+
+    it('should update model when order changes', function() {
+      inject(function($compile, $rootScope) {
+        var element;
+        element = $compile('<ul ui-sortable ng-model="items"><div class="dummy"></div><li ng-repeat="item in items" id="s-{{$index}}">{{ item }}</li><div class="dummy"></div></ul>')($rootScope);
```

Comment:
Sure @thgreasi . Maybe we should wrap all the specs in `sortable.e2e.spec.js` in a function, e.g. `function runTests(beforeElement, afterElement)` that will run all the specs, with `beforeElement` added before the `ul` element and `afterElement` after it.
We could than avoid the current duplication of code by doing

```js
runTests(); // run the tests with no added elements
runTests('<li></li>', '<li></li>'); // run the tests with added elements
```